rep-0155.rst

```diff
@@ -0,0 +1,902 @@
+   perform human perception/social signal recognition per se.  It only specify
+   naming conventions and interfaces between these nodes.
+
+The use of capitalized words such as MUST, SHOULD, MAY must be understood as
+specified in the IETF RFC2119 'Key words for use in RFCs to Indicate Requirement
+Levels' [6]_.
+
+Human Representation
+====================
+
+To accommodate existing tools and technique used to detect and recognise
+humans, the representation of a person is build on a combination of 4
+unique identifiers (UUIDs): a **person identifier**, a **face identifier**, a
+**body identifier** and a **voice identifier**. Future revisions of this REP
+might add additional identifiers.
+
+These four identifiers are not mutually exclusive, and depending on the
+requirements of the application, the available sensing capabilities, and the
+position/behaviour of the humans, only some might be available for a given
+person, at a given time.
+
+Person Identifier
+-----------------
+
+The **person identifier** MUST be a unique ID (typically, a UUID) permanently
+associated with a unique person. This person ID is normally assigned by a node
+able to perform person identification (face recognition node, voice
+recognition node, sound source localisation + name, identification based on
+physical features like height/age/gender, person identification based on
+pre-defined features like the colour of the clothes, etc.) This ID is meant to
+be **persistent** so that the robot can recognise people across
+encounters/sessions. Nodes providing person IDs MAY serialise these IDs to a
+permanent storage, for them to persist across robot reboots.
+
+When meaningful (see section `Person frame`_), a TF frame MUST be
+associated to the person ID and named ``person_<personID>``. Due to the
+importance of the head in human-robot interaction, the ``person_<personID>``
+frame is expected to be placed as close as possible to the head of the human.
+If neither the face nor the skeleton are tracked, the ``person_<personID>``
+frame might be located to the last known position of the human, or removed
+altogether if no meaningful estimate of the human location is available. See
+section `Person frame`_ for details regarding the ``person_<personID>`` frame.
+
+At any given time, the list of known persons is published under the
+``/humans/persons/tracked`` topic as ``hri_msgs/IdsList`` messages.
+
+Merging Person Identifiers
+''''''''''''''''''''''''''
+
+In certain cases, two person IDs must be merged (for instance, the robot
+detects that a voice and a face that were thought to belong to different people
+are indeed the same person).
+
+In such a case, one of the person ID is marked as an *alias* of the other
+person, by publishing the ID of the other person on a special subtopic named
+``alias``. See section `Topics structure`_ for details.
+
+.. Note:: The reverse operation (spliting a person into two) can be realised
```

Comment:
Hi - I am not very familiar with the ROS standardization process, but I went over the document linked here: [REP-155-20220713.pdf](https://github.com/ros-infrastructure/rep/files/9103731/REP.155.--.Conventions.Topics.Interfaces.for.Perception.in.Human-Robot.Interaction.ROS.org.pdf) and I do not see what @clalancette sees, namely that this is a particular HRI implementation. These seem to be useful and extremely generic conventions for any person-interactive robotic system. 

On a more general note, I think that having conventions and data structures that HRI ROS folks agree on would greatly enhance adoption of standard libraries and tools that could save HRI researchers time and accelerate research.

Reply:
In fairness to clancette and the OSRF folks, it _is_ a particular HRI implementation ([here](https://github.com/ros4hri/ros4hri))... the question is "is this a general enough toolset that should be made standard across all of ROS, or is it a standalone tool"? Generally standards like this would be more "here's what a message should look like", this standard is focused on what functionality will be available to a system: it's more of an attempt at standardizing an architecture.

Personally, while I do feel that it describes a specific implementation, I also feel that that implementation prioritizes being general and is therefore useful for a wide array of use-cases.

Reply:
As I said, I might not be looking in the right place, but the REP PDF linked above only (mostly?) describes what the messages should be, and not how they are implemented. Does the REP also include a specific reference to the implementation you linked to?

Reply:
@cst0 as per ROS process, [REP must have a reference implementation](https://www.ros.org/reps/rep-0001.html#rep-work-flow) -- that is what you found under https://github.com/ros4hri/

As pointed by @guyhoffman , the REP itself is design to be agnostic of a particular implementation -- of course, one might always object that a specification always shape implementations to some extend...

In any case, I'd be happy to try to address specific issue with the spec that over-constrain potential implementations.

Reply:
(as a side note, I'm already aware of a 2nd implementation of the spec that only reuse one or two nodes from the reference implementation)